File: next/kmp/core/src/commonMain/kotlin/org/dweb_browser/core/ipc/helper/IpcError.kt
package org.dweb_browser.core.ipc.helper

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@SerialName(IPC_MESSAGE_TYPE_ERROR)
data class IpcError(
  val errorCode: Int,
  val message: String? = null,
) : IpcRawMessage, IpcMessage {
  override fun toString() = "IpcError(errorCode=$errorCode,message=$message)"
}

File: desktop-dev/src/core/ipc/IpcError.ts
import { IPC_MESSAGE_TYPE } from "./helper/const.ts";
import { IpcMessage } from "./helper/IpcMessage.ts";

export class IpcError extends IpcMessage<IPC_MESSAGE_TYPE.ERROR> {
  constructor(readonly errorCode: number, readonly message?: string) {
    super(IPC_MESSAGE_TYPE.ERROR);
  }

  static internalServer(message: string) {
    return new IpcError(500, message);
  }
}
